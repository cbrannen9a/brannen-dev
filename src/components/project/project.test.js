import React from 'react';
import renderer from 'react-test-renderer';

import Project from './project';

const mock = {
    title: 'Test Project',
    url: 'testurl',
    imageData: {
        aspectRatio: 1.5,
        src: 'test_image.jpg',
        srcSet: 'some srcSet',
        srcSetWebp: 'some srcSetWebp',
        sizes: '(max-width: 600px) 100vw, 600px',
        base64: 'string_of_base64'
    },
    body: []
};

describe('Project', () => {
    it('renders correctly', () => {
        const tree = renderer.create(<Project {...mock} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
