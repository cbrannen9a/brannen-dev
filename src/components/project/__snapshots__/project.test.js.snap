// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project renders correctly 1`] = `
<div
  className="project"
>
  <h1
    className="title"
  >
    Test Project
  </h1>
  <div
    className=" gatsby-image-wrapper"
    style={
      Object {
        "overflow": "hidden",
        "position": "relative",
      }
    }
  >
    <div
      style={
        Object {
          "paddingBottom": "66.66666666666667%",
          "width": "100%",
        }
      }
    />
    <img
      alt=""
      src="string_of_base64"
      style={
        Object {
          "height": "100%",
          "left": 0,
          "objectFit": "cover",
          "objectPosition": "center",
          "opacity": 1,
          "position": "absolute",
          "top": 0,
          "transitionDelay": "500ms",
          "width": "100%",
        }
      }
    />
    <picture>
      <source
        sizes="(max-width: 600px) 100vw, 600px"
        srcSet="some srcSetWebp"
        type="image/webp"
      />
      <source
        sizes="(max-width: 600px) 100vw, 600px"
        srcSet="some srcSet"
      />
      <img
        alt="Test Project"
        loading="lazy"
        onLoad={[Function]}
        sizes="(max-width: 600px) 100vw, 600px"
        src="test_image.jpg"
        srcSet="some srcSet"
        style={
          Object {
            "height": "100%",
            "left": 0,
            "objectFit": "cover",
            "objectPosition": "center",
            "opacity": 0,
            "position": "absolute",
            "top": 0,
            "transition": "opacity 500ms",
            "width": "100%",
          }
        }
      />
    </picture>
    <noscript
      dangerouslySetInnerHTML={
        Object {
          "__html": "<picture><source type='image/webp' srcset=\\"some srcSetWebp\\" sizes=\\"(max-width: 600px) 100vw, 600px\\" /><source srcset=\\"some srcSet\\" sizes=\\"(max-width: 600px) 100vw, 600px\\" /><img loading=\\"lazy\\" sizes=\\"(max-width: 600px) 100vw, 600px\\" srcset=\\"some srcSet\\" src=\\"test_image.jpg\\" alt=\\"Test Project\\" style=\\"position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center\\"/></picture>",
        }
      }
    />
  </div>
  
  <p>
    <a
      className="button is-link"
      href="testurl"
    >
      View this project online →
    </a>
  </p>
  <p>
    <a
      className="button is-light"
      href="/"
    >
      ← back to all projects
    </a>
  </p>
</div>
`;
